aws/resource_aws_kinesis_analytics_application.go

```diff
@@ -0,0 +1,1521 @@
+											Schema: map[string]*schema.Schema{
+												"mapping": {
+													Type:     schema.TypeString,
+													Optional: true,
+												},
+
+												"name": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+
+												"sql_type": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+											},
+										},
+									},
+
+									"record_encoding": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+
+									"record_format": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"mapping_parameters": {
+													Type:     schema.TypeList,
+													Optional: true,
+													MaxItems: 1,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"csv": {
+																Type:     schema.TypeList,
+																Optional: true,
+																MaxItems: 1,
+																Elem: &schema.Resource{
+																	Schema: map[string]*schema.Schema{
+																		"record_column_delimiter": {
+																			Type:     schema.TypeString,
+																			Required: true,
+																		},
+
+																		"record_row_delimiter": {
+																			Type:     schema.TypeString,
+																			Required: true,
+																		},
+																	},
+																},
+															},
+
+															"json": {
+																Type:     schema.TypeList,
+																Optional: true,
+																MaxItems: 1,
+																Elem: &schema.Resource{
+																	Schema: map[string]*schema.Schema{
+																		"record_row_path": {
+																			Type:     schema.TypeString,
+																			Required: true,
+																		},
+																	},
+																},
+															},
+														},
+													},
+												},
+
+												"record_format_type": {
+													Type:     schema.TypeString,
+													Computed: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+
+						"table_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsKinesisAnalyticsApplicationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kinesisanalyticsconn
+	name := d.Get("name").(string)
+	createOpts := &kinesisanalytics.CreateApplicationInput{
+		ApplicationName: aws.String(name),
+	}
+
+	if v, ok := d.GetOk("code"); ok && v.(string) != "" {
+		createOpts.ApplicationCode = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("cloudwatch_logging_options"); ok {
+		clo := v.([]interface{})[0].(map[string]interface{})
+		cloudwatchLoggingOption := createCloudwatchLoggingOption(clo)
```

Comment:
Since this project uses a very flat Go package structure, we prefer to include the AWS service name in the function name. We also generally call these functions "expand" functions. e.g. `expandKinesisAnalyticsCloudwatchLoggingOption`

Same for the below `createX` function names.